packages/relic/lib/src/headers/special/cookie_header.dart

```diff
@@ -0,0 +1,68 @@
+part of '../headers.dart';
+
+/// A class representing the HTTP Cookie header.
+///
+/// This class manages the parsing and representation of cookies.
+class CookieHeader {
+  /// The list of cookies.
+  final List<Cookie> cookies;
+
+  /// Constructs a [CookieHeader] instance with the specified cookies.
+  CookieHeader({required this.cookies});
+
+  /// Parses the Cookie header value and returns a [CookieHeader] instance.
+  ///
+  /// This method processes the header value, extracting the cookies into a list.
+  factory CookieHeader.parse(String value) {
+    final splitValues = value.splitTrimAndFilterUnique(separator: ';');
+
+    if (splitValues.isEmpty) {
+      throw FormatException('Value cannot be empty');
+    }
+
+    final cookies = splitValues.map((pair) {
+      final parts = pair.split('=');
+      if (parts.length != 2) {
+        throw FormatException('Invalid cookie format');
+      }
+
+      return Cookie(name: parts[0].trim(), value: parts[1].trim());
+    }).toList();
+
+    return CookieHeader(cookies: cookies);
+  }
+
+  Cookie? getCookie(String name) {
+    return cookies.firstWhere((cookie) => cookie.name == name);
+  }
+
+  /// Converts the [CookieHeader] instance into a string representation suitable for HTTP headers.
+  String toHeaderString() {
+    return cookies.map((cookie) => cookie.toHeaderString()).join('; ');
+  }
+
+  @override
+  String toString() {
+    return 'CookieHeader(cookies: $cookies)';
+  }
+}
+
+/// A class representing a single cookie.
+class Cookie {
+  /// The name of the cookie.
+  final String name;
+
+  /// The value of the cookie.
+  final String value;
+
+  /// Constructs a [Cookie] instance with the specified name and value.
+  Cookie({required this.name, required this.value});
```

Comment:
Should we validate that the Strings are valid?

Cookie Name:

Must not contain:
Any whitespace characters (space, tabs, etc.).
Any of the following characters: ()<>@,;:\"/[]?={} \t.
Generally, the name should only contain:
Alphanumeric characters (A-Z, a-z, 0-9).
Special characters like _, -, or ..
Cookie Value:

Can contain:
Alphanumeric characters (A-Z, a-z, 0-9).
Special characters like !#$%&'*+-.^_|~\`.
Encoded characters like %20 (for spaces).
Must not contain:
Control characters (ASCII 0–31 or 127).
Reserved characters without being encoded (e.g., \`;

Reply:
This may apply to other headers, too.

Reply:
The rule for these are here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

Reply:
@vlidholt fixed here: https://github.com/serverpod/serverpod/pull/2744/commits/cc673d4935d8cf1f34bd53317c16c607ab86cd7e